### PR TITLE
Tel 85 define tenant source model

### PR DESCRIPTION
**Pull request by NagarajNidagundi**

### CODE REVIEW

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: b221ac67de2af8a54cecd67dd31cb757b2e97288
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=601e259d-ff3b-4d4d-a40b-7f0ad45f4dd1%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: 4b954b4ec312c5119fe2c90444fa05ce9418b642
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4517195a-7854-4290-87c6-f174e0856441%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: 4b954b4ec312c5119fe2c90444fa05ce9418b642
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4517195a-7854-4290-87c6-f174e0856441%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: 4b954b4ec312c5119fe2c90444fa05ce9418b642
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4517195a-7854-4290-87c6-f174e0856441%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.SourceService/Models/SourceContext.cs [hunk 1]
```diff
@@ -54,7 +59,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
             modelBuilder.Entity<Source>(entity =>
             {
-                entity.ToTable("source").HasKey(e => e.Id);
+                entity.ToTable("source").HasKey(e => e.PublicId);
```
Comment (davidw-foxtrax) [inline src/WWTelematics.SourceService/Models/SourceContext.cs]: Why is this changing?  I would expect id to remain the primary key.

File: src/WWTelematics.SourceService/Models/SourceContext.cs [hunk 2]
```diff
@@ -75,6 +80,37 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 entity.Property(e => e.CreatedTime)
                    .HasColumnName("created_time").IsRequired();
             });
+
+            modelBuilder.Entity<TenantSource>(entity =>
+            {
+                entity.ToTable("tenant_source").HasKey(ts => new { ts.TenantId, ts.SourceId });
+
+                entity.Property(e => e.Id)
```
Comment (davidw-foxtrax) [inline src/WWTelematics.SourceService/Models/SourceContext.cs]: Id should still be the primary key

File: src/WWTelematics.SourceService/Models/SourceContext.cs (hunk 1 above)
Comment (NagarajNidagundi) [reply src/WWTelematics.SourceService/Models/SourceContext.cs]: image

By keeping id as the primary key, Im unable to run the migration

Comment (NagarajNidagundi) [reply src/WWTelematics.SourceService/Models/SourceContext.cs]: migartion error
By keeping id as the primary key, Im unable to run the migration

**Comment by davidw-foxtrax** [comment]
To address the key issue - 

I think the best way to proceed is to work similarly to what we did in client.  The value saved in the database is the integer id and foreign keyed to the primary key integer id of tenant.  The value we expose to the users, though, is the public id of the tenant.

Does that sound reasonable?

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: a7a02ec83eb3b5b0982ad756f1e04b4b00b4abae
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4b40f7aa-d74e-4550-a022-8223196b3dd3%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsSourceService-CodeBuildProject
* Commit ID: cc0b271aa134925906cc198beb6c757fe1342ad1
* Result: FAILED
* [Build Logs](https://9lxt94z6ka.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d0195c63-9bb6-49cf-9967-6701096d0040%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.SourceService/Models/SourceContext.cs (hunk 2 above)
Comment (NagarajNidagundi) [reply src/WWTelematics.SourceService/Models/SourceContext.cs]: done

**Comment by davidw-foxtrax** [comment]
In this case, the TenantId should still be a Guid, because it is referencing a value outside the microservice.

The Guid SourceId still needs to be exposed for everything outside the microservice, with the int id only used internally as the foreign key for easier mapping.

**Comment by davidw-foxtrax** [comment]
Replaced by #19

**Pull request closed without merging**